astropy/cosmology/io/html.py

```diff
@@ -0,0 +1,103 @@
+        is a string, the "name" column is used as the indexing column.
+
+    move_to_meta : bool (optional, keyword-only)
+        Whether to move keyword arguments that are not in the Cosmology class'
+        signature to the Cosmology's metadata. This will only be applied if the
+        Cosmology does NOT have a keyword-only argument (e.g. ``**kwargs``).
+        Arguments moved to the metadata will be merged with existing metadata,
+        preferring specified metadata in the case of a merge conflict
+        (e.g. for ``Cosmology(meta={'key':10}, key=42)``, the ``Cosmology.meta``
+        will be ``{'key': 10}``).
+
+    cosmology : str, `~astropy.cosmology.Cosmology` class, or None (optional, keyword-only)
+        The cosmology class (or string name thereof) to use when constructing
+        the cosmology instance. The class also provides default parameter values,
+        filling in any non-mandatory arguments missing in 'table'.
+
+    **kwargs
+        Passed to :attr:`astropy.table.QTable.read`
+
+    Returns
+    -------
+    `~astropy.cosmology.Cosmology` subclass instance
+    """
+    kwargs["format"] = "ascii.html"
+    with u.add_enabled_units(cu):
+        table = QTable.read(filename, **kwargs)
+
+    # build cosmology from table
+    return from_table(table, index=index, move_to_meta=move_to_meta, cosmology=cosmology)
+
+def write_table(cosmology, file, *, overwrite=False, cls=QTable, format_dict=None, cosmology_in_meta=True, **kwargs):
+    """Serialize the cosmology into a HTML table.
+
+    Parameters
+    ----------
+    cosmology : `~astropy.cosmology.Cosmology` subclass instance
+    file : path-like or file-like
+        Location to save the serialized cosmology.
+
+    overwrite : bool
+        Whether to overwrite the file, if it exists.
+    cls : type (optional, keyword-only)
+        Astropy :class:`~astropy.table.Table` (sub)class to use when writing.
+        Default is :class:`~astropy.table.QTable`.
+    cosmology_in_meta : bool
+        Whether to put the cosmology class in the Table metadata (if `True`,
+        default) or as the first column (if `False`).
+    **kwargs
+        Passed to ``cls.write``
+
+    Raises
+    ------
+    TypeError
+        If kwarg (optional) 'cls' is not a subclass of `astropy.table.Table`
+    """
+    table = to_table(cosmology, cls, cosmology_in_meta=False)
+
+    if cls == QTable:
+        cosmo_cls = type(Planck18)
+        for n, col in zip(table.colnames, table.itercols()):
+            if n != "cosmology":
+                param = getattr(cosmo_cls, n)  # Get parameter, and check it; fails here
+                if not isinstance(param, Parameter) or param.unit in (None, u.one):
+                    continue
+                # Replace column with unitless version
+                table.replace_column(n, col.to_value(param.unit), copy=False)
+
+    if format_dict is not None and type(format_dict) is dict:

```

Comment:
I like the format_dict, but maybe this would be better as a variable in the file with preloaded values than as a function argument.
It will make reading much easier if column renaming were the same every time. Also, it will be easier for the user to not have column-by-column options.

Reply:
I like the idea of having preloaded values in the HTML. I am looking in the documentation about whether astropy has built-in methods to store metadata and preloaded values separate apart from within the HTML table and preloading values. Do you have any suggestions? 

I think it should be possible to store the metadata and preloaded values as their own tables or as json data separate from the HTML table, and I imagine there might be something in astropy that does that already.

Reply:
Apologies. What I meant was for ``format_table`` to be a dictionary in the file and not function argument.
So before ``def write_table(...):`` there would be

```python
format_dict = {"H0": "$H_0$", ...}
```

And `write_table` would use this.

Reply:
Ah I see. So `format_table` should be a variable outside of any function within `html.py`? In that case, I should add some getters and setters to `html.py`.

Reply:
As an update, I updated the branch to have the format table outside of write_table(). I also added code in read_table to account for translated names to revert back to their original form. I will add the test cases once I am done writing them.

It does look like trying to store metadata outside of the HTML table does not appear to work, since the test fails to pass. I am still looking as to why, but I am not sure that the option is available based on what I see so far.